src/main/java/net/sf/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -304,26 +315,58 @@ public void saveAs() throws Throwable {
             }
         }
 
-        File oldFile = panel.getBibDatabaseContext().getDatabaseFile().orElse(null);
-        panel.getBibDatabaseContext().setDatabaseFile(file);
+        BibDatabaseContext context = panel.getBibDatabaseContext();
+
+        if (context.getLocation() == DatabaseLocation.SHARED) {
+            // Generate an ID when saving a shared database
+            String databaseID = new BigInteger(128, new SecureRandom()).toString(32);
+            context.getDatabase().setDatabaseID(databaseID);
```

Comment:
The database should take care of its ID and thus also how a new one is generated -> move this two lines to a new method `createNewID` in the database class. I would also rename everything there from databaseID to simply ID (since it is already a member of the database class so it should be clear from the context)